backend/ops_api/ops/resources/agreements.py

```diff
@@ -271,6 +271,9 @@ def delete(self, id: int) -> Response:
                 elif any(bli.status != BudgetLineItemStatus.DRAFT for bli in agreement.budget_line_items):
                     raise RuntimeError(f"Agreement {id} has budget line items not in draft status.")
 
+                for bli in agreement.budget_line_items:
```

Comment:
should we catch any specific conditional errors that might occur during the BLI deletion, that would result in the Agreement not getting deleted? (thought process, if something goes wrong during the BLI.delete, then we probably shouldn't continue on deleting the Agreement (orphaned BLIs at that point). I presume the `try` would capture that, but it doesn't look explicit.

open to thoughts